airflow/stats.py

```diff
@@ -348,25 +348,28 @@ def timer(self, stat=None, *args, tags=None, **kwargs):
 
 
 class _Stats(type):
+    factory = None
     instance: Optional[StatsLogger] = None
 
     def __getattr__(cls, name):
+        if not cls.instance:
+            try:
+                cls.instance = cls.factory()
+            except (socket.gaierror, ImportError) as e:
+                log.error("Could not configure StatsClient: %s, using DummyStatsLogger instead.", e)
```

Comment:
If a stats backend _isn't_ configured, isn't factory going to be None, and so we'd die with a TypeError?

Reply:
Oh no, cos this is a metaclass.